tests/rancher-istio/rancher_istio.go

```diff
@@ -0,0 +1,947 @@
+				SetValue(
+					"installer.repository", "testRepo/mirrored-istio-installer",
+				).
+				SetValue(
+					"installer.tag", "1.7.3",
+				),
+		},
+
+		{
+			Name: "Set .Values.installer.releaseMirror and .Values.installer.debug.secondsSleep",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"installer.releaseMirror.enabled", "true",
+				).
+				SetValue(
+					"installer.debug.secondsSleep", "30",
+				),
+		},
+
+		{
+			Name: "Set CNI to true",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"cni.enabled", "true",
+				),
+		},
+
+		{
+			Name: "Set CNI to false",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"cni.enabled", "false",
+				),
+		},
+
+		{
+			Name: "Set CNI repository and tag",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"cni.repository", "testRepo/mirrored-istio-install-cni",
+				).
+				SetValue(
+					"cni.tag", "1.7.3",
+				),
+		},
+
+		{
+			Name: "Set .Values.cni.logLevel and .Values.cni.excludeNamespaces",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"cni.logLevel", "debug",
+				).
+				SetValue(
+					"cni.excludeNamespaces", "testNamespace",
+				).
+				SetValue(
+					"cni.excludeNamespaces", "testNamespace2",
+				),
+		},
+
+		{
+			Name: "Set .Values.pilot.enabled to true",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"pilot.enabled", "true",
+				),
+		},
+
+		{
+			Name: "Set .Values.pilot.enabled to false",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"pilot.enabled", "false",
+				),
+		},
+
+		{
+			Name: "Set .Values.pilot.repository and .Values.pilot.tag",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"pilot.repository", "testRepo/mirrored-istio-pilot",
+				).
+				SetValue(
+					"pilot.tag", "1.7.3",
+				),
+		},
+
+		{
+			Name: "Set .Values.pilot.hpaSpec",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"pilot.hpaSpec", testHPASpec,
+				),
+		},
+		{
+			Name: "Set .Values.pilot.podDisruptionBudget",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"pilot.podDisruptionBudget", testPodDisruptionBudget,
+				),
+		},
+
+		{
+			Name: "Set .Values.egressGateways.enabled to true",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"egressGateways.enabled", "true",
+				),
+		},
+
+		{
+			Name: "Set .Values.egressGateways.enabled to false",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
```

Comment:
Should these `egressGateways` test cases be checked together or is it fine if they're seperate?

Reply:
ah right. merged them into one case

Reply:
If you could squash these into one `.Set()` i think it would be cleaner. 

i.e.
```
  Set(
    "egressGateways", map[string]interface{}{
    "enabled":             false,
    "type":                "NodePort",
    "hpaspec":             testHPASpec,
    "podDisruptionBudget": testPodDisruptionBudget,
  }),
```

Reply:
If you could, please do this in other places where it applies as well